src/intersections/segments.jl

```diff
@@ -37,9 +40,9 @@ function intersection(f, seg₁::Segment{N,T}, seg₂::Segment{N,T}) where {N,T}
     elseif (λc == l₁ && λd > l₁) || (λd == l₁ && λc > l₁)
       return @IT CornerTouching b f # CASE 3
     else
-      params = sort([0, 1, λc / l₁, λd / l₁])
-      p₁ = seg₁(params[2])
-      p₂ = seg₁(params[3])
+      o₁, o₂ = overlapparameters(zero(T), one(T), λc / l₁, λd / l₁)
```

Comment:
rename to _sort4vals or another low-level internal name and define this function in the same source file as opposed to moving it to utils.jl